packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.jsx

```diff
@@ -349,6 +349,48 @@ const DataSeriesFormItem = ({
     });
   };
 
+  const dataItemListItems = dataSection?.map((dataItem, i) => {
+    const iconColorOption =
+      dataItem.color || DATAITEM_COLORS_OPTIONS[i % DATAITEM_COLORS_OPTIONS.length];
+    return {
+      id: dataItem.dataSourceId,
+      content: {
+        value: dataItem.label || dataItem.dataSourceId,
```

Comment:
This actually needs to prioritize the dataItemId before the dataSourceId since the dataSourceId will be generated with a uuid. `dataItem.label || dataItem.dataItemId || dataItem.dataSourceId`

Also, you'll need to make this change in `TableCardFormContent` as well.

Reply:
We probably should mark `dataItemId` as required in the dataItem PropType

Reply:
Made both dataSourceId and dataItemId required